cli/src/main/java/com/nextdoor/bender/Bender.java

```diff
@@ -0,0 +1,227 @@
+      logger.error(e.getMessage());
+      formatter.printHelp(name, options);
+      System.exit(1);
+    }
+
+    // The CLI tool doesn't have any configuration files built into it. We
+    // require that the user set BENDER_CONFIG to something reasonable.
+    if (System.getenv("BENDER_CONFIG") == null) {
+      logger.error(
+              "You must set the BENDER_CONFIG environment variable. \n" +
+              "Valid options include: file://<file>");
+      formatter.printHelp(name, options);
+      System.exit(1);
+    }
+
+    if (cmd.hasOption("help")) {
+      formatter.printHelp(name, options);
+      System.exit(0);
+    }
+
+    // Depending on the desired Handler, we invoke a specific method and pass in the
+    // options (or defaults) required for that handler.
+    String handler_value = cmd.getOptionValue(handler.getLongOpt(), KINESIS);
+    try {
+
+      switch (handler_value.toLowerCase()) {
+
+        // Note: We only support Kinesis right now. Will add support for S3/S3SNS down the road.
+        case KINESIS:
+          invokeKinesisHandler(
+                  cmd.getOptionValue(kinesis_stream_name.getLongOpt(), KINESIS_STREAM_NAME),
+                  cmd.getOptionValue(source_file.getLongOpt())
+          );
+          break;
+
+        // Error out if an invalid handler was supplied.
+        default:
+          logger.error("Invalid Handler Option (" + handler_value + "), valid options are: " + KINESIS);
+          formatter.printHelp(name, options);
+          System.exit(1);
+      }
+    } catch (HandlerException e) {
+      logger.error("Error executing handler: " + e);
+      System.exit(1);
+    }
+  }
+
+  protected static void invokeKinesisHandler(String stream_name, String source_file) throws HandlerException {
+    String sourceArn = "arn:aws:kinesis:" + AWS_REGION + ":" + AWS_ACCOUNT + ":stream/" + stream_name;
+    logger.info("Invoking the Kinesis Handler...");
+
+    TestContext ctx = getContext();
+    KinesisHandler handler = new KinesisHandler();
+
+    // Set the arrival timestamp as the function run time.
+    Date approximateArrivalTimestamp = new Date();
+    approximateArrivalTimestamp.setTime(System.currentTimeMillis());
+
+    // Open up the source file for events
+    Scanner scan = null;
+    try {
+      scan = new Scanner(new File(source_file));
+    } catch (FileNotFoundException e) {
+      logger.error("Could not find source file (" + source_file + "): " + e);
+      System.exit(1);
+    }
+
+    // Create a series of KinesisEvents from the source file. All of these events will be treated
+    // as a single batch that was pushed to Kinesis, so they will all have the same Arrival Time.
+    logger.info("Parsing " + source_file + "...");
+    List<KinesisEvent.KinesisEventRecord> events = new ArrayList<KinesisEvent.KinesisEventRecord>();
+    int r = 0;
+
+    // Walk through the source file. For each line in the file, turn the line into a KinesisRecord.
+    while(scan.hasNextLine()){
```

Comment:
You could argue here that we add a `max_records` setting and then create multiple `KinesisEvent` objects and pass each one into the Handler. Thats an optimization that might save on memory and give a more realistic behavior .. but seemed unnecessary in v1.